src/Microsoft.OData.Core/JsonLight/ODataJsonLightValueSerializer.cs

```diff
@@ -330,6 +330,16 @@ public virtual void WriteNullValue()
         {
             Debug.Assert(value != null, "value != null");
 
+#if NETCOREAPP3_1_OR_GREATER
+            if (value is ODataJsonElementValue jsonElementValue)
+            {
+                // We don't perform validation for ODataJsonElementValue.
+                // We assume the content is valid.

```

Comment:
Where do you get such assumptions?

Reply:
This assumption is part of the design choices behind the `ODataJsonElementValue` and is documented here: https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/Value/ODataJsonElementValue.cs#L24

Essentially, the benefit of using `ODataJsonElementValue` is to pass-through `JsonElement` values without having to parse them. Since the `ODataWriter` does not parse nor inspect the contents of the `ODataJsonElementValue`, we do not attempt to validate it.

Reply:
I think this should be publicly documented. What do you think?

Reply:
Technically it is publicly documented since doc comments end up in the public reference documented page for that class. That said, we should include an entry in the main docs as well.